qa-pipelines/tasks/cf-upgrade.sh

```diff
@@ -98,7 +98,48 @@ wait_for_namespace() {
 
 PROVISIONER=$(kubectl get storageclasses persistent -o "jsonpath={.provisioner}")
 
-# Deploy UAA
+# monitor_url takes a URL argument and a path to a log file
+# This will time out after 30 hours. Until then, repeatedly curl the URL with a 1-second wait period, and log the response
+# If the application state changes, print this to stdout as well
+monitor_url() {
+  local app_url=$1
+  local log_file=$2
+  local count=0
+  local last_state=
+  local new_state=
+  echo "monitoring URL ${app_url}"
+  while true; do
+    new_state=$({ curl -sI "${1}" || echo "URL could not be reached"; } | head -n1 | tee -a "${log_file}")
+    if [[ "${new_state}" != "${last_state}" ]]; then
+      echo "state change for ${1}: ${new_state}"
```

Comment:
(suggestion) 
```monitoring URL http://go-env.10.84.101.213.omg.howdoi.website
state change for http://go-env.10.84.101.213.omg.howdoi.website: HTTP/1.1 200 OK
Release "uaa" has been upgraded. Happy Helming!```

it will be nice if we `last_state` is also shown
` echo "state change for ${1}: ${new_state} from ${last_state}"`

Reply:
I can't think of an easy way to say this that will make sense with the first state change (when last state is empty), without introducing additional logic around that. Anyway, the last state can be seen by scrolling back to the previous announcement of state change.